Batch of Code Review docs refactoring (#291)
submariner-io/submariner-website: src/content/contributing/code-review/_index.en.md

## Code Review Guide

This guide is meant to facilitate Submariner code review by sharing norms, best practices, and useful patterns.

Submariner follows the [Kubernetes Code Review Guide][kube code review guide] wherever relevant. This guide collects the most important
highlights of the K8s process and adds Submariner-specific extensions.

### Two non-author Committer approvals required

Pull Requests to Submariner require two approvals from a Committer to the relevant part of the code base, as defined by the CODEOWNERS file
at the root of each repository and the [Community
Membership/Committers](http://localhost:1313/contributing/community-membership/#committers) process.

### No merge commits

Kubernetes recommends [avoiding merge commits][merge commits].

> Our current GitHub configuration should automatically remove merge commits.

Reviewer: ```suggestion
Our current GitHub configuration should automatically remove merge commits.
```

With our current GitHub setup, pull requests are liable to include merge commits temporarily: whenever a PR is updated through the UI, GitHub merges the target branch into the PR. However, since we merge PRs by either squashing or rebasing them, those merge commits disappear from the series of commits which ultimately ends up in the target branch.

## Code Review Guide

This guide is meant to facilitate Submariner code review by sharing norms, best practices, and useful patterns.

Submariner follows the [Kubernetes Code Review Guide][kube code review guide] wherever relevant. This guide collects the most important
highlights of the K8s process and adds Submariner-specific extensions.

### Two non-author Committer approvals required

Pull Requests to Submariner require two approvals from a Committer to the relevant part of the code base, as defined by the CODEOWNERS file
at the root of each repository and the [Community
Membership/Committers](http://localhost:1313/contributing/community-membership/#committers) process.

### No merge commits

Kubernetes recommends [avoiding merge commits][merge commits].

> With our current GitHub setup, pull requests are liable to include merge commits temporarily: whenever a PR is updated through the UI,

Reviewer 2: ```suggestion
With our current GitHub setup, pull requests are liable to include merge commits temporarily. Whenever a PR is updated through the UI,
```

GitHub merges the target branch into the PR. However, since we merge PRs by either squashing or rebasing them, those merge commits
disappear from the series of commits which ultimately ends up in the target branch.

### Squash/amend commits into discrete steps

Kubernetes recommends [squashing commits using these guidelines][squashing].

After a review, prepare your PR for merging by squashing your commits.

All commits left on your branch after a review should represent meaningful milestones or units of work. Use commits to add clarity to the

> development and review process.

Reviewer 2: To further expand:

```suggestion
development and review process. Keep in mind that smaller commits are easier to review. In addition. when addressing review comments, as a general rule, push a new commit instead of amending to the prior commit as the former makes it easy for reviewers to determine what changed.
```

Reviewer: Following on Tom’s suggestion: 

```
But remember to squash them into the appropriate commit before merging. To make that simpler, commit with `git commit --fixup`, with the appropriate commit hash: this will keep your review-prompted changes as separate commits, but automatically select them for squashing with `git rebase --autosquash -i`.
```

GitHub merges the target branch into the PR. However, since we merge PRs by either squashing or rebasing them, those merge commits
disappear from the series of commits which ultimately ends up in the target branch.

### Squash/amend commits into discrete steps

Kubernetes recommends [squashing commits using these guidelines][squashing].

After a review, prepare your PR for merging by squashing your commits.

All commits left on your branch after a review should represent meaningful milestones or units of work. Use commits to add clarity to the
development and review process. Keep in mind that smaller commits are easier to review.

Before merging a PR, squash the following kinds of commits:

* Fixes/review feedback
* Typos
* Merges and rebases
* Work in progress
* Aim to have every commit in a PR compile and pass tests independently if you can, but it's not a requirement.

### Address code review feedback with new commits

When addressing review comments, as a general rule, push a new commit instead of amending to the prior commit as the former makes it easy
for reviewers to determine what changed.

> But remember to squash them into the appropriate commit before merging. To make that simpler, commit with `git commit --fixup`, with the
> appropriate commit hash: this will keep your review-prompted changes as separate commits, but automatically select them for squashing with
> `git rebase --autosquash -i`.

Reviewer 2: Not trying to nit but expanded the content a bit - hope it's all accurate.

```suggestion
To avoid cluttering the git log, squash the review commits into the appropriate commit before merging. The committer can do this in GitHub via the "Squash and merge" option.  However you may want to preserve other commits, in which case, squashing will need to be done manually via the Git CLI. To make that simpler, you can commit the review-prompted changes with `git commit --fixup` with the
appropriate commit hash. This will keep them as separate commits and, when you later rebase, specify the `--autosquash` option (that is `git rebase --autosquash -i`) to automatically select them for squashing.
```
